### DISCUSSION TITLE
Rust Coding Guidelines Doc

### DISCUSSION THREAD

**Discussion by zeeshanlakhani** [fission-codes/rust-template#7] [General]
What are some of the patterns we should `recommend` for programming in Rust, other than adhering to typical project layout, test separation (unit vs integration), clippy, & fmt? 

Some more info: https://rust-lang.github.io/api-guidelines/about.html

**Comment by BoisterousCoder** [comment]
I was thinking we could recommend when possible trying to pass in borrows like `&str` or `&mut my_struct` and then pass out owned variables like `String`. This obviously doesn't make sense in a lot of cases but it makes it easier for people to use said functions.
